extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/pages/databaseSettingsPage.ts

```diff
@@ -128,6 +135,31 @@ export class DatabaseSettingsPage extends BasePage {
 		this._endIpAddressTextRow = this.wizard.createFormRowComponent(view, constants.EndIpAddressLabel, '', this._endIpAddressTextbox, true);
 	}
 
+	private createFirewallToggle(view: azdata.ModelView) {
+		this._firewallToggleDropdown = view.modelBuilder.dropDown().withProps({
+			values: [
+				{
+					displayName: localizedConstants.yes,
+					name: 'True'
+				},
+				{
+					displayName: localizedConstants.no,
+					name: 'False'
+				}
+			]
+		}).component();
+
+		this._firewallToggleDropdown.onValueChanged((value) => {
+			let dropDownValue = (this._firewallToggleDropdown.value as azdata.CategoryValue).name;
+			let displayValue: 'block' | 'none' = dropDownValue === 'True' ? 'block' : 'none';
+			this.wizard.changeRowDisplay(this._firewallRuleNameTextRow, displayValue);
+			this.wizard.changeRowDisplay(this._endIpAddressTextRow, displayValue);
+			this.wizard.changeRowDisplay(this._startIpAddressTextRow, displayValue);
+			this.wizard.changeComponentDisplay(this._IpInfoText, displayValue);
+			this.wizard.model.newFirewallRule = dropDownValue;
```

Comment:
I'd suggest converting this to a boolean here instead - easier to work with for the stuff that uses the model value